ocs_ci/helpers/helpers.py

```diff
@@ -6001,3 +6002,118 @@ def find_cephfilesystemsubvolumegroup(storageclient_uid=None):
         cephbfssubvolumegroup = storage_consumer.get_cephfs_subvolumegroup()
 
     return cephbfssubvolumegroup
+
+
+def get_rbd_daemonset_csi_addons_node_object(node):
+    """
+    Gets rdb daemonset CSI addons node data
+
+    Args:
+        node (str): Name of the node
+
+    Returns:
+       dict: CSI addons node object info
+
+    """
+    namespace = config.ENV_DATA["cluster_namespace"]
+    csi_addons_node = OCP(kind=constants.CSI_ADDONS_NODE_KIND, namespace=namespace)
+    csi_addons_node_data = csi_addons_node.get(
+        resource_name=f"{node}-{namespace}-daemonset-csi-rbdplugin"
+    )
+    return csi_addons_node_data
+
+
+def create_network_fence_class():
+    """
+    Create NetworkFenceClass CR and verify Ips are populated
+    in respective CsiAddonsNode objects
+
+    """
+
+    logger.info("Creating NetworkFenceClass")
+    network_fence_class_dict = templating.load_yaml(constants.NETWORK_FENCE_CLASS_CRD)
+    network_fence_class_obj = create_resource(**network_fence_class_dict)
+    if network_fence_class_obj.ocp.get(
+        resource_name=network_fence_class_obj.name, dont_raise=True
+    ):
+        logger.info(
+            f"NetworkFenceClass {network_fence_class_obj.name} created successfully"
+        )
+
+    logger.info("Verifying CsiAddonsNode object for CSI RBD daemonset")
+    all_nodes = get_worker_nodes()
+
+    for node_name in all_nodes:
+        cidrs = get_rbd_daemonset_csi_addons_node_object(node_name)["status"][
+            "networkFenceClientStatus"
+        ][0]["ClientDetails"][0]["cidrs"]
+        assert len(cidrs) == 1, "No cidrs are populated to CSI Addons node object"
+        logger.info(f"Cidr: {cidrs[0]} populated in {node_name} CSI addons node object")
+
+
+def create_network_fence(node_name, cidr):
+    """
+    Create NetworkFence for the node
+
+    Args:
+        node_name (str): Name of the node
+        cidr (str): cidr
+
+    Returns:
+        OCS: NetworkFence object
```

Comment:
I guess it shoud be "network_fence_obj" instead of OCS

Reply:
@avd-sagare No its OCS object of kind NetworkFence